Throwable toString override

daddykotex opened the pull request:
Default `toString` on `Throwable` prints the classname. This is suboptimal when we could have the case class `toString` implementation available.

## PR Checklist (not all items are relevant to all PRs)

- [ ] Added unit-tests (for runtime code)
- [x] Update bootstrapped code + smoke tests (when the rendering logic is modified)
- [ ] Added build-plugins integration tests (when reflection loading is required at codegen-time)
- [ ] Added alloy compliance tests (when simpleRestJson protocol behaviour is expanded/updated)
- [ ] Updated dynamic module to match generated-code behaviour
- [ ] Added documentation
- [x] Updated changelog

daddykotex committed:
Override the default toString when extends Throwable

daddykotex committed:
Update examples

daddykotex committed:
Add changelog entry

Baccata commented on modules/codegen/src/smithy4s/codegen/internals/Renderer.scala:
```diff
@@ -801,6 +802,10 @@ private[internals] class Renderer(compilationUnit: CompilationUnit) { self =>
 
+  // Default toString on Throwable prints the classname, instead we use the case class toString.
+  private def defaultThrowableGetMessage =
+    line"override def toString(): $string_ = $scalaRuntime._toString(this)"
```
Sorry for reviewing this only now. Anyway, the documentation of the ScalaRunTime object states the following : 

```scala
/** The object ScalaRunTime provides support methods required by
 *  the scala runtime.  All these methods should be considered
 *  outside the API and subject to change or removal without notice.
 */
```

therefore, I'm not comfortable with this implementation. Can you elaborate on the rationale for wanting to use this ? I guess it's just for the convenience of it. 

In any case, we can render an actual implementation that'd be using a StringBuilder quite easily.

daddykotex replied:
I missed that completely, that's fair.

> In any case, we can render an actual implementation that'd be using a StringBuilder quite easily.

Any format we prefer? For `Foo(bar: String)`, default implementation renders `Foo(barValue)`, we can choose how to render if we implement it ourselves

daddykotex committed:
Revert "Override the default toString when extends Throwable"

This reverts commit 831d99cf535a3e7d3c0e41fe8bad2c5db4983f3d.

daddykotex committed:
Merge branch 'series/0.18' into dfrancoeur/0.18-throwable-tostring

 Conflicts:
	CHANGELOG.md

daddykotex committed:
Render custom toString

I chose to implement toString, because implementing getMessage would
lead to `smithy4s.example.ClientError: smithy4s.example.ClientError(400, "oops")`
which felt weird
